src/main/java/emissary/util/io/FileManipulator.java

```diff
@@ -21,10 +22,14 @@
      */
     private static String FS = System.getProperty("file.separator", "/");
 
+    // Use ThreadLocal to allocate a SecureRandom instance per thread instead
+    // of Math.random() which uses the same instance of Random for all threads
+    // that causes resource contention and is not as secure.
+    private static final ThreadLocal<SecureRandom> secureRandomThreadLocal = ThreadLocal.withInitial(() -> new SecureRandom());
+
     public static String mkTempFile(final String dirPath, final String file) {
-        // Date now = new Date(System.currentTimeMillis());
-        String theFullPath = "";
-        String tempName = file + (int) (Math.rint(Math.random() * 10000));
+        String theFullPath;
+        String tempName = file + secureRandomThreadLocal.get().nextInt(10000);
```

Comment:
Replace with ```nextLong()```

Reply:
Done. Just had to mask the first bit to keep the number positive.